staging/kos/pkg/controllers/subnet/validator.go

```diff
@@ -46,36 +47,32 @@ import (
 const subnetVNIField = "spec.vni"
 
 // conflictsCache holds information for one subnet regarding conflicts with
-// other subnets. There's no guarantee that the cache is up-to-date: a subnet Y
-// stored in it might no longer be in conflict with the owning subnet, for
-// instance because of an update to Y's CIDR following its addition to the
-// cache.
+// other subnets. There's no guarantee that the cache is up-to-date. For
+// instance, if a conflicting  subnet X is in it and gets deleted it is not
+// removed.
 type conflictsCache struct {
-	// ownerSummary stores the data relevant to validation for the subnet owning
-	// the conflicts cache.
-	ownerSummary *subnet.Summary
-
-	// rivals identifies the subnets that have a conflict with the subnet owning
-	// the cache. If subnet X is the owner of the conflicts cache and a subnet Y
-	// is in rivals this means that when Y was last processed by a queue worker
-	// a conflict with X was found and at that time X had the VNI and CIDR
-	// values in ownerSummary.
+	// ownerUID is the UID of the subnet owning the conflicts cache.
+	ownerUID k8stypes.UID
```

Comment:
Adding a subnet X to another subnet Y’s conflicts cache was conditional on Y’s VNI and CIDR recorded in the conflicts cache matching those in the Y received from the API server. Since after immutability Y’s VNI and CIDR can change only if Y’s UID changes, compare UIDs instead of VNI and CIDR --- it’s less fine-grained but simpler.